generators/pkg/parser/parser.go

```diff
@@ -3,24 +3,27 @@
 package parser
 
 import (
+	"fmt"
+	"generators/pkg/generator"
 	"io"
+	"io/ioutil"
 
-	"k8s.io/apimachinery/pkg/util/yaml"
+	"sigs.k8s.io/yaml"
 )
 
-type products struct {
-	Items []int    `json:"items,omitempty"`
-	Names []string `json:"names,omitempty"`
-}
-
 // Parse parses a yaml file from io.Reader and stores the
 // result in the products struct
-func Parse(r io.Reader) (products, error) {
-	reader := yaml.NewYAMLToJSONDecoder(r)
-	res := products{}
-	err := reader.Decode(&res)
+func Parse(r io.Reader) (generator.GitHubSpec, error) {
+	res := generator.GitHubSpec{}
```

Comment:
It's generally good practice to declare vars as close to where they are used. In this case, we're declaring res, but then doing nothing with in in the next statement. This would make more sense to be moved down to where the unmarshal occurs.

Reply:
I got it. But here because if `ioutil.ReadAll(r)` has an error, an empty struct and the error will be returned. I could change the return value to be `generator.GitHubSpec{}` instead of the var.

Reply:
Ah, I see! SGTM then :) (either way is fine)